### PR TITLE
adding srt/1.4.1

### PR DESCRIPTION
**Pull request by prince-chrismc**
Specify library name and version:  **srt/1.4.1**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by prince-chrismc** [comment]
There's a new line tab.......

File: recipes/srt/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    description = "Secure Reliable Transport (SRT) is an open source transport technology that optimizes streaming performance across unpredictable networks, such as the Internet."
+    topics = ("conan", "srt", "ip", "transport")
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "MPL-2.0"
+    exports_sources = ["CMakeLists.txt", "patches/*"]
+    generators = "cmake"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+    short_paths = True
```
[the remark this answers is not in the document]
Comment (prince-chrismc) [reply recipes/srt/all/conanfile.py:17]: It failed on CCI because of path/file name too long

[the remark this answers is not in the document]
Comment (uilianries) [reply recipes/srt/all/conanfile.py:17]: @jgsogo said the infra team will enable long path for Windows jobs :)

[the remark this answers is not in the document]
Comment (prince-chrismc) [reply recipes/srt/all/conanfile.py:17]: That's amazing! However consumer will most likely still require this when building locally :worried:

[the remark this answers is not in the document]
Comment (uilianries) [reply recipes/srt/all/conanfile.py:17]: so you are saying that's really required for users? If Yes, I see no problem keeping this in the recipe;

[the remark this answers is not in the document]
Comment (prince-chrismc) [reply recipes/srt/all/conanfile.py:17]: :100: somone will run into problems

`C:\Users\chris\.conan\data\srt\1.4.1\_\_\build\99915f9d64a8a9c9e16f006abf14b94db41e3074\build_subfolder\source_subfolder\srt_virtual.dir\Release\srtsuppo.3E132FEE.tlog\srtsupport_virtual.lastbuildstate`

is 201 characters, thats only 44 more

but that seems pretty standard for a conan build =? I think it's just as problematic as any other!

[the remark this answers is not in the document]
Comment (jgsogo) [reply recipes/srt/all/conanfile.py]: Yes, Windows nodes should have the short-paths limitation disabled (although I'm investigating a failure in PR https://github.com/conan-io/conan-center-index/pull/1931), nevertheless, we should keep adding the `short_paths=True` if needed and at some point in time we should implement this kind of hook: https://github.com/conan-io/conan-center-index/pull/1931

File: recipes/srt/all/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,12 @@
+cmake_minimum_required(VERSION 2.8.11)
+project(cmake_wrapper)
+
+include(conanbuildinfo.cmake)
+conan_basic_setup()
+
+if(CONAN_LIBS_PTHREADS4W)
+    set(PTHREAD_LIBRARY "${CONAN_LIBS_PTHREADS4W}")
+    set(PTHREAD_INCLUDE_DIR "${CONAN_INCLUDE_DIRS_PTHREADS4W}")
+endif()
```
Comment (jgsogo) [inline recipes/srt/all/CMakeLists.txt:10]: Is this a common pattern in other recipes? Is it possible to `find_package(pthreads4w)` and use the provided targets?

File: recipes/srt/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+class SttConan(ConanFile):
+    name = "srt"
+    homepage = "https://github.com/Haivision"
+    description = "Secure Reliable Transport (SRT) is an open source transport technology that optimizes streaming performance across unpredictable networks, such as the Internet."
+    topics = ("conan", "srt", "ip", "transport")
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "MPL-2.0"
+    exports_sources = ["CMakeLists.txt", "patches/*"]
+    generators = "cmake"
```
Comment (jgsogo) [inline recipes/srt/all/conanfile.py:13]:
```suggestion
    generators = "cmake", "cmake_find_package"
```

File: recipes/srt/all/patches/001-cmake-link-targets.patch [hunk 4]
```diff
@@ -0,0 +1,23 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 7fafe77cc..d5c4d701e 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -623,7 +623,7 @@ if (srt_libspec_shared)
+ 	set_target_properties (${TARGET_srt}_shared PROPERTIES VERSION ${SRT_VERSION} SOVERSION ${SRT_VERSION_MAJOR})
+ 	list (APPEND INSTALL_TARGETS ${TARGET_srt}_shared)
+ 	if (ENABLE_ENCRYPTION)
+-		target_link_libraries(${TARGET_srt}_shared PRIVATE ${SSL_LIBRARIES})
++		target_link_libraries(${TARGET_srt}_shared PRIVATE ${CONAN_LIBS_OPENSSL})
```
Comment (jgsogo) [inline recipes/srt/all/patches/001-cmake-link-targets.patch:10]: If we are patching the original files, let's use the targets provided by the requirements

```suggestion
+		target_link_libraries(${TARGET_srt}_shared PRIVATE OpenSSL::OpenSSL)
```

File: recipes/srt/all/patches/001-cmake-link-targets.patch [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+ 	endif()
+ 	if (MICROSOFT)
+ 		target_link_libraries(${TARGET_srt}_shared PRIVATE ws2_32.lib)
+@@ -657,7 +657,7 @@ if (srt_libspec_static)
+ 
+ 	list (APPEND INSTALL_TARGETS ${TARGET_srt}_static)
+ 	if (ENABLE_ENCRYPTION)
+-		target_link_libraries(${TARGET_srt}_static PRIVATE ${SSL_LIBRARIES})
++		target_link_libraries(${TARGET_srt}_static PRIVATE ${CONAN_LIBS_OPENSSL})
```
Comment (jgsogo) [inline recipes/srt/all/patches/001-cmake-link-targets.patch:19]:
```suggestion
+		target_link_libraries(${TARGET_srt}_static PRIVATE OpenSSL::OpenSSL)
```

File: recipes/srt/all/test_package/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,10 @@
+cmake_minimum_required(VERSION 3.1.2)
+project(test_package CXX)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup(TARGETS)
+
+find_package(srt REQUIRED)
+
+add_executable(${CMAKE_PROJECT_NAME} test_package.cpp)
+target_link_libraries(${CMAKE_PROJECT_NAME} CONAN_PKG::srt)
```
Comment (jgsogo) [inline recipes/srt/all/test_package/CMakeLists.txt:10]:
```suggestion
target_link_libraries(${CMAKE_PROJECT_NAME} srt::srt)
```

File: recipes/srt/all/CMakeLists.txt (hunk 2 above)
Comment (prince-chrismc) [reply recipes/srt/all/CMakeLists.txt:10]: `pthreads4w` does not have CMake support so there's no expectation from consumer to find it. This project provides options for providing them... they could be moved to the recipe under `_configure_cmake()` however I do not know how to extract these settings from cpp_info.

**Comment by prince-chrismc** [comment]
> I'm suggesting some changes: if we are using targets, let's use them in other places if possible.

The only tricky part is this project supports many "SSL backends" and my intention is to only add support for OpenSSL.

I am trying to get permission to ask the SRT Alliance to improve their CMake, since it does not follow many of the common practices.

**Comment by prince-chrismc** [comment]
> I'm suggesting some changes: if we are using targets, let's use them in other places if possible.

I forgot I had tried that in the first place

File: recipes/srt/all/conanfile.py [hunk 7]
```diff
@@ -0,0 +1,71 @@
+from conans import ConanFile, CMake, tools
+import os
+
+
+class SttConan(ConanFile):
```
Comment (SSE4) [inline recipes/srt/all/conanfile.py:5]:
```suggestion
class SrtConan(ConanFile):
```

File: recipes/srt/all/conanfile.py [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+    name = "srt"
+    homepage = "https://github.com/Haivision"
```
Comment (SSE4) [inline recipes/srt/all/conanfile.py:7]:
```suggestion
    homepage = "https://github.com/Haivision/srt"
```

File: recipes/srt/all/test_package/test_package.cpp [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+
+   memset(&hints, 0, sizeof(struct addrinfo));
+   hints.ai_flags = AI_PASSIVE;
+   hints.ai_family = AF_INET;
+   hints.ai_socktype = SOCK_DGRAM;
+
+   std::string service("9000");
+   if (0 != getaddrinfo(NULL, service.c_str(), &hints, &res))
+   {
+      return 1;
```
Comment (SSE4) [inline recipes/srt/all/test_package/test_package.cpp:30]:
```suggestion
      srt_cleanup();
      return 1;
```

File: recipes/srt/all/test_package/test_package.cpp [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+   }
+
+   SRTSOCKET serv = srt_socket(res->ai_family, res->ai_socktype, res->ai_protocol);
+   srt_close(serv);
+
+   // use this function to release the UDT library
+   srt_cleanup();
```
Comment (SSE4) [inline recipes/srt/all/test_package/test_package.cpp:37]:
```suggestion
   freeaddrinfo(res);
   srt_cleanup();
```

**Comment by prince-chrismc** [comment]
```  Checking Build System

C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\IDE\VC\VCTargets\Microsoft.CppCommon.targets(181,5): error MSB4018: The "GetOutOfDateItems" task failed unexpectedly. [C:\J\workspace\onan-center-pull-request_PR-2126@2\10\pr_2126_10_0_10\.conan\data\srt\1.4.1\_\_\build\99915f9d64a8a9c9e16f006abf14b94db41e3074\build_subfolder\source_subfolder\haicrypt_virtual.vcxproj]

C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\IDE\VC\VCTargets\Microsoft.CppCommon.targets(181,5): error MSB4018: System.InvalidOperationException: The item metadata "%(FullPath)" cannot be applied to the path "haicrypt_virtual.dir\Release\haicrypt_virtual.tlog\ParallelCustomBuild.command.1.tlog". Path: haicrypt_virtual.dir\Release\haicrypt_virtual.tlog\ParallelCustomBuild.command.1.tlog exceeds the OS max path limit. The fully qualified file name must be less than 260 characters. [C:\J\workspace\onan-center-pull-request_PR-2126@2\10\pr_2126_10_0_10\.conan\data\srt\1.4.1\_\_\build\99915f9d64a8a9c9e16f006abf14b94db41e3074\build_subfolder\source_subfolder\haicrypt_virtual.vcxproj]

C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\IDE\VC\VCTargets\Microsoft.CppCommon.targets(181,5): error MSB4018:    at Microsoft.Build.Shared.ErrorUtilities.ThrowInvalidOperation(String resourceName, Object[] args) [C:\J\workspace\onan-center-pull-request_PR-2126@2\10\pr_2126_10_0_10\.conan\data\srt\1.4.1\_\_\build\99915f9d64a8a9c9e16f006abf14b94db41e3074\build_subfolder\source_subfolder\haicrypt_virtual.vcxproj]

C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\IDE\VC\VCTargets\Microsoft.CppCommon.targets(181,5): error MSB4018:    at Microsoft.Build.Shared.FileUtilities.ItemSpecModifiers.GetItemSpecModifier(String currentDirectory, String itemSpec, String definingProjectEscaped, String modifier, String& fullPath) [C:\J\workspace\onan-center-pull-request_PR-2126@2\10\pr_2126_10_0_10\.conan\data\srt\1.4.1\_\_\build\99915f9d64a8a9c9e16f006abf14b94db41e3074\build_subfolder\source_subfolder\haicrypt_virtual.vcxproj]

C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\IDE\VC\VCTargets\Microsoft.CppCommon.targets(181,5): error MSB4018:    at Microsoft.Build.Utilities.TaskItem.Microsoft.Build.Framework.ITaskItem2.GetMetadataValueEscaped(String metadataName) [C:\J\workspace\onan-center-pull-request_PR-2126@2\10\pr_2126_10_0_10\.conan\data\srt\1.4.1\_\_\build\99915f9d64a8a9c9e16f006abf14b94db41e3074\build_subfolder\source_subfolder\haicrypt_virtual.vcxproj]

C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\IDE\VC\VCTargets\Microsoft.CppCommon.targets(181,5): error MSB4018:    at Microsoft.Build.Utilities.TaskItem.GetMetadata(String metadataName) [C:\J\workspace\onan-center-pull-request_PR-2126@2\10\pr_2126_10_0_10\.conan\data\srt\1.4.1\_\_\build\99915f9d64a8a9c9e16f006abf14b94db41e3074\build_subfolder\source_subfolder\haicrypt_virtual.vcxproj]

C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\IDE\VC\VCTargets\Microsoft.CppCommon.targets(181,5): error MSB4018:    at Microsoft.Build.CPPTasks.GetOutOfDateItems.Execute() [C:\J\workspace\onan-center-pull-request_PR-2126@2\10\pr_2126_10_0_10\.conan\data\srt\1.4.1\_\_\build\99915f9d64a8a9c9e16f006abf14b94db41e3074\build_subfolder\source_subfolder\haicrypt_virtual.vcxproj]

C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\IDE\VC\VCTargets\Microsoft.CppCommon.targets(181,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [C:\J\workspace\onan-center-pull-request_PR-2126@2\10\pr_2126_10_0_10\.conan\data\srt\1.4.1\_\_\build\99915f9d64a8a9c9e16f006abf14b94db41e3074\build_subfolder\source_subfolder\haicrypt_virtual.vcxproj]

C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\IDE\VC\VCTargets\Microsoft.CppCommon.targets(181,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.<ExecuteInstantiatedTask>d__26.MoveNext() [C:\J\workspace\onan-center-pull-request_PR-2126@2\10\pr_2126_10_0_10\.conan\data\srt\1.4.1\_\_\build\99915f9d64a8a9c9e16f006abf14b94db41e3074\build_subfolder\source_subfolder\haicrypt_virtual.vcxproj]

  Building Custom Rule C:/J/workspace/onan-center-pull-request_PR-2126@2/10/pr_2126_10_0_10/.conan/data/srt/1.4.1/_/_/build/99915f9d64a8a9c9e16f006abf14b94db41e3074/source_subfolder/CMakeLists.txt
```

CI having trouble?

**Comment by jgsogo** [comment]
It looks like the long path issue is bitting us again. Here the same problem: https://github.com/conan-io/conan-center-index/pull/1931

**Comment by jgsogo** [comment]
The path in the CI workspaces is a little bit shorter with the `short_paths=True`... I've found that Windows OS is not the only one with problems related to _long paths_, it is MSBuild itself: https://github.com/microsoft/msbuild/issues/53#issuecomment-459062618

This PR works, but others are blocked. We need to use a workspace directory as short as possible.

**Comment by prince-chrismc** [comment]
> The path in the CI workspaces is a little bit shorter with the `short_paths=True`

I am curious what you mean by this, my naive view is that this

```
C:\J\workspace\onan-center-pull-request_PR-2126@2\10\pr_2126_10_0_10\.conan\data\srt\1.4.1\_\_\build\99915f9d64a8a9c9e16f006abf14b94db41e3074
```
becomes
`C:\.conan\data\srt\1.4.1\_\_\build\99915f9d64a8a9c9e16f006abf14b94db41e3074`
removing
`J\workspace\onan-center-pull-request_PR-2126@2\10\pr_2126_10_0_10`

which is almost half. :thinking:

**Comment by jgsogo** [comment]
> > The path in the CI workspaces is a little bit shorter with the `short_paths=True`
> 
> I am curious what you mean by this, my naive view is that this
> 
> ```
> C:\J\workspace\onan-center-pull-request_PR-2126@2\10\pr_2126_10_0_10\.conan\data\srt\1.4.1\_\_\build\99915f9d64a8a9c9e16f006abf14b94db41e3074
> ```
> 
> becomes
> `C:\.conan\data\srt\1.4.1\_\_\build\99915f9d64a8a9c9e16f006abf14b94db41e3074`
> removing
> `J\workspace\onan-center-pull-request_PR-2126@2\10\pr_2126_10_0_10`
> 
> which is almost half. 🤔

Not exactly. If you have a look to the logs you will see that the short path is something like 

```
C:\J\workspace\onan-center-pull-request_PR-2126@2/13/pr_2126_13_0_105/.short\63d6ea\1\build_subfolder
```

Having it all inside the workspace folder for each job makes it easier to manage in the CI. I thought that the only problem was with Windows OS and long paths so I requested the infrastructure team to change the registry entry to remove that limitation and we moved the short_path folder inside the workspace one. Now I know that MSBuild also has this issue even if Windows OS doesn't... 😢 

Next move will be to keep everything inside the workspace folder, but use a shorter path for it. I'll propose something like `C:\J\ws\PR2126@2/13/105/.short/...`  removing everything that is not strictly needed

**Comment by prince-chrismc** [comment]
> Not exactly. If you have a look to the logs you will see that the short path is something like (we assign it via env variable)

I'll need to search for that environment variable! tyvm for the details. I can see there is some bloat in the path comparing your too examples.

Looking at the logs, I had a good laugh when I noticed they are all missing a 'c' for 'conan-center'

```
C:\J\workspace\onan-center-pull-request_PR-2126@2/13/pr_2126_13_0_105/.short\63d6ea\1\build_subfolder
              ^^
```

**Comment by jgsogo** [comment]
> 
> Looking at the logs, I had a good laugh when I noticed they are all missing a 'c' for 'conan-center'
> 
> ```
> C:\J\workspace\onan-center-pull-request_PR-2126@2/13/pr_2126_13_0_105/.short\63d6ea\1\build_subfolder
>               ^^
> ```

it is one of those things that everyone knows, but no one is fixing it 🤷  😆  😓 (not sure about the proper reaction, hahaha)

File: recipes/srt/all/test_package/CMakeLists.txt (hunk 6 above)
Comment (jgsogo) [inline recipes/srt/all/test_package/CMakeLists.txt:10]: Let's use the target provided by the `find_package(srt)` call

```suggestion
target_link_libraries(${CMAKE_PROJECT_NAME} srt::srt)
```

**Pull request merged**